### PR TITLE
[BRO] implement Prototype

### PR DESCRIPTION
**Pull request by Susucre**
Hello,
This is my take on implementing the Prototype Ability from Brother's War.

I went with introducing a prototyped boolean for both Permanents and Spells. That is set to true upon casting a spell with a prototype SpellAbility then propagated on cast, resolve, copy (both copy on stack & battlefield) and clone.
Then, a few continuousEffect are added through PrototypeActivity, that conditionnaly set Color, ManaCost, and base Power/Toughness for potentially prototyped cards.

The only weird behavior I've noticed testing the ability is that on cancelling the cast of a prototyped spell, the card wrongly display the prototyped power/toughness back in the hand, but that does not look to alter how the cast behave from there.

### CODE REVIEW

**Comment by Susucre** [comment]
[[Blitz Automaton]]

Also, solves part of BRO set tracking #9598

File: Mage/src/main/java/mage/abilities/keyword/PrototypeAbility.java [hunk 1]
```diff
@@ -1,17 +1,68 @@
...
  */
 public class PrototypeAbility extends SpellAbility {
 
-    public PrototypeAbility(Card card, String manaString, int power, int toughness) {
-        super(new ManaCostsImpl<>(manaString), card.getName());
-        // TODO: implement this
+    private int power;
+    private int toughness;
+
+    public PrototypeAbility(Card card, String manaString, ObjectColor color, int power, int toughness) {
```
Comment (theelk801) [inline Mage/src/main/java/mage/abilities/keyword/PrototypeAbility.java:30]: why do we need to pass in color here? the mana cost should be enough info for that

**Review by theelk801 — CHANGES_REQUESTED** [review]
would you be able to write tests for this? it needs a pretty comprehensive test suite

**Comment by Susucre** [comment]
Hey there!
Sorry I got no time at all to dedicate to the project right now.
Both unit tests and deriving the prototype color from the manacost are worth additions.
Feel free to take over the branch or start fresh, I will not have time for improving this until at least June.

**Comment by theelk801** [comment]
Hey, thanks for responding! I'll probably take over this branch. Thanks for all the work you've done so far.

**Comment by theelk801** [comment]
switching this over to #10280

**Pull request closed without merging**